### PR TITLE
BC-6352 - Deleting the creator field causes the files not to be displayed anymore

### PR DESCRIPTION
**Pull request by bn-pass**
…ters to see files created by them) in homeworks

# Description

## Links to Tickets or other pull requests

https://ticketsystem.dbildungscloud.de/browse/BC-6352

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by blazejpass — APPROVED** [review]

**Review by dyedwiper — APPROVED** [review]

File: controllers/homework.js
Comment (SevenWaysDP) [inline controllers/homework.js:602]: Why do you remove it completly? What is with other user? For example:
you has removed a Student, but a Teacher or other Student look this submission wath than?

**Review by SevenWaysDP — CHANGES_REQUESTED** [review]

File: controllers/homework.js
Comment (bn-pass) [reply controllers/homework.js:602]: @SevenWaysDP It just removes a requirement for the creator to be present among the person that can access the file which can be sometimes not true as we have to remove the creator reference from the files. The files should remain accessible to everyone that have access to it.

Comment (SevenWaysDP) [reply controllers/homework.js:602]: Sorry @bn-pass but I would like to test this again briefly. There are so many scenarios

Comment (SevenWaysDP) [reply controllers/homework.js:602]: image

there is a problem

Comment (SevenWaysDP) [reply controllers/homework.js:602]: image

Comment (dyedwiper) [reply controllers/homework.js:602]: Interesting! Then these checks were apparently there to distinguish between submission and grading files. Is there another way to distinguish them? As they have both have parentType="submission", this is a real problem, I guess.

Comment (SevenWaysDP) [reply controllers/homework.js:602]: yes

Comment (dyedwiper) [reply controllers/homework.js:602]: Yes, there is another way, or, yes, this is a real problem? ;)

Comment (bn-pass) [reply controllers/homework.js:602]: @SevenWaysDP where did you spot that behaviour? Was it on the testing environment for this branch or on your local development machine and you only checked out `schulcloud-client`? I'm afraid it could be related to the changes made to the "authorization" logic in the backend (on the same branch)... I don't think changes made here (in the `schulcloud-client`) could cause that behaviour as I've only removed the check for the creator presence among the `submitters` or `teachers` (as we still want to be able to access some files uploaded by no longer "present" user.

Comment (SevenWaysDP) [reply controllers/homework.js:602]: https://bc-6352-no-creator-ref-case.nbc.dbildungscloud.dev/tasks

Comment (SevenWaysDP) [reply controllers/homework.js:602]: > Yes, there is another way, or, yes, this is a real problem? ;)

 This is a real problem!

Comment (SevenWaysDP) [reply controllers/homework.js:602]: > @SevenWaysDP where did you spot that behaviour? Was it on the testing environment for this branch or on your local development machine and you only checked out this branch in the `schulcloud-client`? I'm afraid it could be related to the changes made to the "authorization" logic in the backend (on the same branch)... I don't think changes made here (in the `schulcloud-client`) could cause that behaviour as I've only removed the check for the creator presence among the `submitters` or `teachers` (as we still want to be able to access some files uploaded by no longer "present" user.

This is a problem if you remove because submission and grading files have parentType = submission and we sort them by creatorId in Frondend

 https://github.com/hpi-schul-cloud/schulcloud-client/pull/3396#discussion_r1489642411

Comment (SevenWaysDP) [reply controllers/homework.js:602]: Please do not merge, I will clarify this with Cedric and Ingwert.

Comment (bn-pass) [reply controllers/homework.js]: @dyedwiper @SevenWaysDP As the ticket has received a higher priority I've modified the solution to a bit safer approach which came from a fact that the issue has not been created by this particular ticket's changes (as it hasn't added the files' creatorId removal - it has been added in another ticket, already merged to main so it will already happen on all the executed data deletion requests). I think it's a sweet spot between what is available right now to us and what we want to achieve in the follow-up ticket (BC-6609).

So thanks to the changes made within the https://github.com/hpi-schul-cloud/schulcloud-client/pull/3396/commits/4e555a63c4b61c8d276170e9aee29b41e9cfef43 commit, the filter will now be applied to all the files that still have the _creatorId_ reference defined, but the files that do not have this reference will simply be added to every tab (will be selected by any filter) - thanks to this any files should not be left unavailable in any case. And after the BC-6609 will be developed and merged this inconvenience will be fixed (but unfortunately not for the old files as they will already have the _creatorId_ reference removed) as we'll then be able to decide whether it's a submission-related file or grading-related file based on some flag or something.

Please re-review all the changes and let me know what's the conclusion - in my opinion it should be enough to not block this PR anymore, but you have the final voice of course.

**Pull request merged**
